scripts/src/javatests/org/oppia/android/scripts/maven/GenerateMavenDependenciesListTest.kt

```diff
@@ -0,0 +1,423 @@
+    "bumptech/glide/annotations/4.11.0/annotations-4.11.0.pom"
+  private val FIREBASE_ANALYTICS_POM = "https://maven.google.com/com/google/firebase/" +
+    "firebase-analytics/17.5.0/firebase-analytics-17.5.0.pom"
+
+  private val THIRD_PARTY_PREFIX = "//third_pary:"
+  private val DATA_BINDING_COORD = "androidx.databinding:databinding-adapters:3.4.2"
+  private val PROTO_LITE_COORD = "com.google.protobuf:protobuf-lite:3.0.0"
+  private val GLIDE_ANNOTATIONS_COORD = "com.github.bumptech.glide:annotations:4.11.0"
+  private val FIREBASE_ANALYTICS_COORD = "com.google.firebase:firebase-analytics:17.5.0"
+  private val IO_FABRIC_COORD = "io.fabric.sdk.android:fabric:1.4.7"
+
+  private val DATA_BINDING_COORD_VERSION = "3.4.2"
+  private val PROTO_LITE_COORD_VERSION = "3.0.0"
+  private val GLIDE_ANNOTATIONS_COORD_VERSION = "4.11.0"
+  private val FIREBASE_ANALYTICS_COORD_VERSION = "17.5.0"
+  private val IO_FABRIC_COORD_VERSION = "1.4.7"
+
+  private val LICENSE_DETAILS_INCOMPLETE_FAILURE = "Licenses details are not completed"
+  private val UNAVAILABLE_OR_INVALID_LICENSE_LINKS_FAILURE =
+    "License links are invalid or not available for some dependencies"
+  private val SCRIPT_PASSED_MESSAGE =
+    "Script executed succesfully: maven_dependencies.textproto updated successfully."
+
+  private val outContent: ByteArrayOutputStream = ByteArrayOutputStream()
+  private val originalOut: PrintStream = System.out
+
+  private val mockLicenseFetcher by lazy { initializeLicenseFetcher() }
+  private lateinit var testBazelWorkspace: TestBazelWorkspace
+
+  @Rule
+  @JvmField
+  var tempFolder = TemporaryFolder()
+
+  @Before
+  fun setUp() {
+    tempFolder.newFolder("scripts", "assets")
+    tempFolder.newFolder("third_party")
+    testBazelWorkspace = TestBazelWorkspace(tempFolder)
+    System.setOut(PrintStream(outContent))
+  }
+
+  @After
+  fun restoreStreams() {
+    System.setOut(originalOut)
+  }
+
+  @Test
+  fun testLicenseLinkNotVerified_scriptFailsWithException() {
+    tempFolder.newFile("scripts/assets/maven_dependencies.pb")
+    tempFolder.newFile("scripts/assets/maven_dependencies.textproto")
+
+    val coordsList = listOf(DATA_BINDING_COORD, FIREBASE_ANALYTICS_COORD)
+    setupBazelEnvironment(coordsList)
+
+    val exception = assertThrows(Exception::class) {
+      GenerateMavenDependenciesListRunner(
+        mockLicenseFetcher,
+        CommandExecutorImpl(600_000L)
+      ).main(
+        arrayOf(
+          "${tempFolder.root}",
+          "scripts/assets/maven_install.json",
+          "${tempFolder.root}/scripts/assets/maven_dependencies.pb"
+        )
+      )
+    }
+    assertThat(exception).hasMessageThat().contains(LICENSE_DETAILS_INCOMPLETE_FAILURE)
+  }
+
+  @Test
+  fun testDependencyDoesNotHaveAnyLicense_scriptFailsWithException() {
+    tempFolder.newFile("scripts/assets/maven_dependencies.textproto")
+    tempFolder.newFile("scripts/assets/maven_dependencies.pb")
+
+    val coordsList = listOf(PROTO_LITE_COORD)
```

Comment:
This could be ``DEP_WITHOUT_LICENSE`` to make it really clear to the reader. Ditto for the other dependencies.

Reply:
Done.

Reply:
Doesn't appear done, but I don't feel strongly about this one.